src/browser/fileImpl.ts

```diff
@@ -0,0 +1,420 @@
+      suggestedName: obj.defaultPath,
+      fileType: { description: "Wave File", accept: { "audio/wav": [".wav"] } },
+    });
+  };
+
+export const showTextSaveDialogImpl: typeof window[typeof SandboxKey]["showTextSaveDialog"] =
+  async (obj: { title: string; defaultPath?: string }) => {
+    return requestSaveFileNameWithDirectoryPermission({
+      suggestedName: obj.defaultPath,
+      fileType: {
+        description: "Text File",
+        accept: {
+          "text/plain": [".txt"],
+        },
+      },
+    });
+  };
+
+export const showOpenDirectoryDialogImpl: typeof window[typeof SandboxKey]["showOpenDirectoryDialog"] =
+  async () => {
+    const _directoryHandler = await showWritableDirectoryPicker();
+    if (_directoryHandler === undefined) {
+      return undefined;
+    }
+
+    await storeDirectoryHandle(_directoryHandler);
+
+    // NOTE: 同一のディレクトリ名だった場合、後で選択されたディレクトリがそれ移行の処理で使用されるため、意図しない保存が発生するかもしれない
+    directoryHandleMap.set(_directoryHandler.name, _directoryHandler);
+    return _directoryHandler.name;
+  };
+
+export const showProjectSaveDialogImpl: typeof window[typeof SandboxKey]["showProjectSaveDialog"] =
+  async (obj: { title: string; defaultPath?: string }) => {
+    return requestSaveFileNameWithDirectoryPermission({
+      suggestedName: obj.defaultPath,
+      fileType: {
+        description: "VOICEVOX Project file",
+        accept: {
+          "application/json": [".vvproj"],
+        },
+      },
+    });
+  };
+
+export const showProjectLoadDialogImpl: typeof window[typeof SandboxKey]["showProjectLoadDialog"] =
+  async () => {
+    return requestLoadFileNameWithDirectoryPermission({
+      fileType: {
+        description: "VOICEVOX Project file",
+        accept: {
+          "application/json": [".vvproj"],
+        },
+      },
+    });
+  };
+
+export const showImportFileDialogImpl: typeof window[typeof SandboxKey]["showImportFileDialog"] =
+  async () => {
+    return requestLoadFileNameWithDirectoryPermission({
+      fileType: {
+        description: "Text",
+        accept: {
+          "text/plain": [".txt"],
+        },
+      },
+    }).then((v) => v?.[0]);
+  };
+
+const isRootPathButDirectoryNotSelected = (path: string) =>
+  directoryHandleMap.get(lastSelectedDirectoryHandleSymbol) === undefined &&
+  path.indexOf(sep) === -1;
+
+const fetchStoredDirectoryHandle = async (maybeDirectoryHandleName: string) => {
+  const db = await openDB();
+  return new Promise<FileSystemDirectoryHandle | undefined>(
+    (resolve, reject) => {
+      const transaction = db.transaction(directoryHandleStoreKey, "readonly");
+      const store = transaction.objectStore(directoryHandleStoreKey);
+      const request = store.get(maybeDirectoryHandleName);
+      request.onsuccess = () => {
+        resolve(request.result);
+      };
+      request.onerror = () => {
+        reject(request.error);
+      };
+    }
+  ).catch(() => {
+    // FIXME: 握り潰してる
+    return undefined;
+  });
```

Comment:
ただエラーを握りつぶすのは利点がないはず！
```suggestion
  );
```